contrib/win32/win32compat/ssh-agent/connection.c

```diff
@@ -134,6 +134,18 @@ process_request(struct agent_connection* con)
 	debug("process agent request type %d", type);
 
 	switch (type) {
+	case SSH_AGENTC_REQUEST_RSA_IDENTITIES:
+	case SSH_AGENT_RSA_IDENTITIES_ANSWER:
+	case SSH_AGENTC_RSA_CHALLENGE:
+	case SSH_AGENT_RSA_RESPONSE:
+	case SSH_AGENT_FAILURE:
+	case SSH_AGENT_SUCCESS:
+	case SSH_AGENTC_ADD_RSA_IDENTITY:
+	case SSH_AGENTC_REMOVE_RSA_IDENTITY:
+	case SSH_AGENTC_REMOVE_ALL_RSA_IDENTITIES:
+		debug("agent request %d: ssh protocol 1 is not supported", type);
+		r = 0;
+		break;
```

Comment:
As per http://api.libssh.org/rfc/PROTOCOL.agent, you'll need to return a response containing SSH_AGENT_FAILURE
`SSH_AGENT_FAILURE messages are also sent in reply to unknown request
types.
`
recommend adding a routine called process_unsupported_request() and populating "response" with SSH_AGENT_FAILURE.